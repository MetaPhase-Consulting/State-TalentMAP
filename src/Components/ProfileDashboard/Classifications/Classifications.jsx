import PropTypes from 'prop-types';
import { difference, isEmpty } from 'lodash';
import FA from 'react-fontawesome';
import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchClassifications, fetchUserClassifications, updateClassifications } from 'actions/classifications';
import { orderClassifications } from 'Components/BidderPortfolio/helpers';
import SectionTitle from '../SectionTitle';
import CheckboxList from '../../BidderPortfolio/CheckboxList';


const Classifications = props => {
  const {
    userId,
    isPublic,
    hideTitle,
    canEditClassifications,
  } = props;

  const dispatch = useDispatch();

  const clientClassifications = useSelector(state => state.userClassificationsSuccess);
  const classifications = useSelector(state => state.classificationsFetchDataSuccess);

  const [editView, setEditView] = useState(false);
  const [userInput, setUserInput] = useState(clientClassifications);

  useEffect(() => {
    dispatch(fetchClassifications());
    dispatch(fetchUserClassifications(userId));
  }, []);

  useEffect(() => {
    setUserInput(clientClassifications);
  }, [clientClassifications]);

  const handleInput = (c) => {
    const pushClass = [...userInput];
    if (!pushClass.includes(c)) {
      pushClass.push(c);
    } else {
      const index = pushClass.indexOf(c);
      if (index > -1) {
        pushClass.splice(index, 1);
      }
    }
    setUserInput(pushClass);
  };

  const cancelInput = () => {
    setUserInput(clientClassifications);
    setEditView(false);
  };

  const onSubmit = () => {
    const updateDiff = {
      insert: difference(userInput, clientClassifications),
      delete: difference(clientClassifications, userInput),
    };
    if (isEmpty(updateDiff.insert) && isEmpty(updateDiff.delete)) {
      setEditView(false);
    } else {
      dispatch(updateClassifications(updateDiff, userId));
    }
  };

  const classifications$ = orderClassifications(classifications);

  return (
    <div className="usa-grid-full profile-section-container updates-container">
      <div className="section-padded-inner-container">
        {
          !hideTitle &&
          <div className="usa-width-one-whole">
            <SectionTitle title="Bidder Classifications" icon="tasks" />
          </div>
        }
        <div className="usa-width-one-whole">
          <CheckboxList
            list={classifications$}
            id="updates"
            editView={editView}
            updateClassifications={handleInput}
            input={userInput}
            isPublic={isPublic}
          />
        </div>
      </div>
      {
        !editView && canEditClassifications &&
        <div className="section-padded-inner-container small-link-container view-more-link-centered">
          <button className="unstyled-button classifications-checkbox" onClick={() => setEditView(true)}>
            <FA
              name="edit"
            />Edit Classifications
          </button>
        </div>
      }
      { editView &&
        <div className="section-padded-inner-container small-link-container view-more-link-centered">
          <div className="saved-search-form-buttons">
            <button
              type="button"
              className="saved-search-form-primary-button"
              onClick={onSubmit}
            >Save
            </button>
            <button
              type="button"
              className="saved-search-form-primary-button"
              onClick={cancelInput}
            >Cancel
            </button>
          </div>
        </div>
      }
    </div>
  );
};

Classifications.propTypes = {
  userId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  isPublic: PropTypes.bool,
  hideTitle: PropTypes.bool,
  canEditClassifications: PropTypes.bool,
};

Classifications.defaultProps = {
  userId: '',
  isPublic: false,
  hideTitle: false,
  canEditClassifications: false,
};

export default Classifications;
