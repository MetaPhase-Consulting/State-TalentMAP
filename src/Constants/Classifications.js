const CLASSIFICATIONS = [
  {
    code: '3',
    text: '3rd Tour Bidders',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: '4',
    text: 'Tenured',
    seasons: [{ id: 215, season_text: 'Summer 2023' }, { id: 214, season_text: 'Summer 2022' }],
  },
  {
    code: 'D',
    text: 'Differential Bidder',
    seasons: [{ id: 215, season_text: 'Summer 2023' }, { id: 214, season_text: 'Summer 2022' }],
  },
  {
    code: 'T',
    text: 'Tandem Bidder',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'M',
    text: 'Meritorious Step Increases',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: '6',
    text: '6/8 Rule',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'F',
    text: 'Fair Share Bidders',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'C',
    text: 'Critical Need Language',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'C1',
    text: 'Critical Need Language 1st Tour Complete',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'CC',
    text: 'Critical Need Language Final Tour Complete',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'R',
    text: 'Recommended for Tenure',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'A',
    text: 'Ambassador or Deputy Assistant Secretary',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'F1',
    text: 'Pickering Fellows',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'F2',
    text: 'Rangel Fellows',
    seasons: [{ id: 185, season_text: null }],
  },
  {
    code: 'P',
    text: 'Pickering/Rangel Fellows',
    seasons: [{ id: 185, season_text: null }],
  },
];

export default CLASSIFICATIONS;
