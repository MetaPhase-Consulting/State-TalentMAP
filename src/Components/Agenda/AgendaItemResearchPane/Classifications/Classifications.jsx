import Classifications from 'Components/ProfileDashboard/Classifications';
import { CLASSIFICATIONS, CLIENT_CLASSIFICATIONS } from 'Constants/PropTypes';

const RemarksGlossary = () => (
  <div className="usa-grid-full classifications-container">
    <Classifications
      userId,
      hideTitle
    />
  </div>
);

RemarksGlossary.propTypes = {
  classifications: CLASSIFICATIONS,
  clientClassifications: CLIENT_CLASSIFICATIONS,
};

RemarksGlossary.defaultProps = {
  classifications: [],
  clientClassifications: [],
};

export default RemarksGlossary;
