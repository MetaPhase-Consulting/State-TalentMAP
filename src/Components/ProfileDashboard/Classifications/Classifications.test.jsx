import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Classifications from './Classifications';

describe('UpdatesComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<Classifications />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Classifications />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
