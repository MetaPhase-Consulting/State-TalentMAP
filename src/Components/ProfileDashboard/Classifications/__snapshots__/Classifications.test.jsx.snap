// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatesComponent matches snapshot 1`] = `
<div
  className="usa-grid-full profile-section-container updates-container"
>
  <div
    className="usa-grid-full section-padded-inner-container"
  >
    <div
      className="usa-width-one-whole"
    >
      <PositionInformation
        className=""
        icon="tasks"
        small={false}
        title="Bidder Classifications"
      />
    </div>
    <div
      className="usa-width-one-whole"
    >
      <CheckboxList
        clientClassifications={Array []}
        id="updates"
        isDisabled={false}
        list={Array []}
      />
    </div>
  </div>
</div>
`;
